File: src/main/java/me/kuku/yuq/logic/QQZoneLogic.java
package me.kuku.yuq.logic;

import com.IceCreamQAQ.Yu.annotation.AutoBind;
import me.kuku.yuq.entity.QQLoginEntity;
import me.kuku.yuq.pojo.Result;

import java.io.IOException;
import java.util.List;
import java.util.Map;

@AutoBind
public interface QQZoneLogic {
    List<Map<String, String>> friendTalk(QQLoginEntity qqLoginEntity) throws IOException;
    List<Map<String, String>> talkByQQ(QQLoginEntity qqLoginEntity, Long qq) throws IOException;
    String forwardTalk(QQLoginEntity qqLoginEntity, String id, String qq, String text) throws IOException;
    String publishTalk(QQLoginEntity qqLoginEntity, String text) throws IOException;
    String removeTalk(QQLoginEntity qqLoginEntity, String id) throws IOException;
    String commentTalk(QQLoginEntity qqLoginEntity, String id, String qq, String text) throws IOException;
    String likeTalk(QQLoginEntity qqLoginEntity, Map<String, String> map) throws IOException;
    Result<List<Map<String, String>>> queryGroup(QQLoginEntity qqLoginEntity) throws IOException;
    Result<List<Map<String, String>>> queryGroupMember(QQLoginEntity qqLoginEntity, String group) throws IOException;
    String leaveMessage(QQLoginEntity qqLoginEntity, Long qq, String content) throws IOException;
    String visitQZone(QQLoginEntity qqLoginEntity, Long qq) throws IOException;
}

File: src/main/kotlin/me/kuku/yuq/logic/QQZoneLogic.kt
package me.kuku.yuq.logic

import com.IceCreamQAQ.Yu.annotation.AutoBind
import me.kuku.yuq.entity.QQLoginEntity
import me.kuku.yuq.pojo.CommonResult
import me.kuku.yuq.pojo.GroupMember

@AutoBind
interface QQZoneLogic {
    //取所有好友说说
    fun friendTalk(qqLoginEntity: QQLoginEntity): List<Map<String, String?>>?
    //取某人说说
    fun talkByQQ(qqLoginEntity: QQLoginEntity, qq: Long): List<Map<String, String?>>?
    //转发说说
    fun forwardTalk(qqLoginEntity: QQLoginEntity, id: String, qq: String, text: String = ""): String
    //发布说说
    fun publishTalk(qqLoginEntity: QQLoginEntity, text: String): String
    //删除说说
    fun removeTalk(qqLoginEntity: QQLoginEntity, id: String): String
    //评论说说
    fun commentTalk(qqLoginEntity: QQLoginEntity, id: String, qq: String, text: String): String
    //点赞说说
    fun likeTalk(qqLoginEntity: QQLoginEntity, map: Map<String, String?>): String
    //发送好友请求   qq号，验证消息，备注，分组名字
    fun addFriend(qqLoginEntity: QQLoginEntity, qq: Long, msg: String, realName: String?, group: String?): String
    //获取所有的群
    fun queryGroup(qqLoginEntity: QQLoginEntity): CommonResult<List<Map<String, String>>>
    //获取群人数
    fun queryGroupMember(qqLoginEntity: QQLoginEntity, group: String): CommonResult<List<Map<String, String>>>
    //留言
    fun leaveMessage(qqLoginEntity: QQLoginEntity, qq: Long, content: String): String
    //访问空间
    fun visitQZone(qqLoginEntity: QQLoginEntity, qq: Long): String
    fun visitQZoneMobile(qqLoginEntity: QQLoginEntity, qq: Long): CommonResult<GroupMember>
}